startup.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import _winreg
+
+from xml.etree import ElementTree
+
+from sgtk import TankError
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+
+class MaxLauncher(SoftwareLauncher):
+    """
+    Handles launching 3dsMax executables. Automatically starts up
+    a tk-3dsmaxplus engine with the current context in the new session
+    of 3dsMax.
+    """
+
+    @property
+    def minimum_supported_version(self):
+        """
+        The minimum software version that is supported by the launcher.
+        """
+        return "2015"
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # First look for executables using the Autodesk Synergy registry.
+        sw_versions = self._synergy_software_versions(versions)
```

Comment:
any reason we are doing both synergy and registry? We were debating this and after some discussion actually ended up dropping the synergy support from Maya. Maybe we could just do the registry based check for max?